### PR TITLE
sql: even more loadgen refactors

### PR DESCRIPTION
**Pull request by mjibson**
available_subsources were being constructed in two places, and the previous refactor missed that (because they happened to produce identical output, but broke when adding TPCH). Refactor to avoid differing implementations.

### Motivation

   * This PR refactors existing code.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  - n/a

### CODE REVIEW

**Comment by mjibson** [comment]
Sigh. One more.

File: src/sql/src/plan/statement/ddl.rs [hunk 1]
```diff
@@ -1059,18 +1043,44 @@ pub fn plan_create_subsource(
...
         }
     };
-    Ok(load_generator)
+
+    let mut available_subsources = HashMap::new();
+    let generator = as_generator(&load_generator);
+    for (i, (name, desc)) in generator.views().iter().enumerate() {
+        let name = FullObjectName {
+            database: RawDatabaseSpecifier::Name("mz_loadgenerator".to_owned()),
+            schema: loadgen.to_string().to_lowercase(),
```
Comment (benesch) [inline src/sql/src/plan/statement/ddl.rs:1069]: If you happen to go back through CI:

```suggestion
            database: RawDatabaseSpecifier::Name("mz_load_generators".to_owned()),
            schema: match loadgen {
                LoadGenerator::Counter => "counter".into(),
                LoadGenerator::Auction => "auction.into(),
                // Please use `snake_case` for any multi-word load generators
                // that you add.
            },
```

If not I'll send it up as a follow on.

**Review by benesch — APPROVED** [review]
Beautiful! I was going to do a similar change myself so I'm glad you beat me to it to avoid the merge conflicts.

File: src/sql/src/plan/statement/ddl.rs (hunk 1 above)
Comment (mjibson) [reply src/sql/src/plan/statement/ddl.rs]: Added.

**Pull request merged**